### PR TITLE
SONARSEC-52 C# "using" statement generates a UCFG block without termi…

### PR DESCRIPTION
**Pull request by valhristov**
…nator

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ public UCFG Build(SyntaxNode syntaxNode, IMethodSymbol methodSymbol)
                 Location = GetLocation(syntaxNode),
             };
 
-            ucfg.BasicBlocks.AddRange(cfg.Blocks.Where(IsSupportedBlock).Select(CreateBasicBlock));
+            ucfg.BasicBlocks.AddRange(cfg.Blocks.Select(CreateBasicBlock));
```
Comment (valhristov) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs]: Adding all blocks now, including Exit; helps with void methods (implicit returns) and exception throws.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UniversalControlFlowGraphBuilder_Blocks.cs [hunk 2]
```diff
@@ -146,9 +311,10 @@ private static void AssertCollection<T>(IList<T> items, params Action<T>[] asser
             }
         }
 
-        private void ValidateJumpBlock(BasicBlock block, string expectedId, params string[] expectedJumps)
+        private void ValidateJmpBlock(BasicBlock block, string expectedId, params string[] expectedJumps)
```
Comment (valhristov) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UniversalControlFlowGraphBuilder_Blocks.cs]: Improves the assert alignment

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs [hunk 3]
```diff
@@ -84,25 +84,22 @@ private BasicBlock CreateBasicBlock(Block block)
...
-                if (block.SuccessorBlocks.All(cfg.ExitBlock.Equals))
-                {
-                    basicBlock.Ret = new Return { ReturnedExpression = ConstantExpression };
-                }
-                else
-                {
-                    basicBlock.Jump = new Jump();
-                    basicBlock.Jump.Destinations.AddRange(block.SuccessorBlocks.Select(blockId.Get));
-                }
+                basicBlock.Ret = new Return { ReturnedExpression = ConstantExpression };
```
Comment (valhristov) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs]: ExitBlock always returns constant. If the method is not void, the corresponding returns will be handled by line 85

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UniversalControlFlowGraphBuilder_Instructions.cs [hunk 4]
```diff
@@ -60,7 +60,7 @@ public void Foo(string s)
 }";
             var ucfg = GetUcfgForMethod(code, "Foo");
 
-            ucfg.BasicBlocks.Should().HaveCount(1);
+            ucfg.BasicBlocks.Should().HaveCount(2);
```
Comment (valhristov) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UniversalControlFlowGraphBuilder_Instructions.cs]: Exit block is now in UCFG, hence the increase of blocks

Comment (benzonico) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UniversalControlFlowGraphBuilder_Instructions.cs]: and things will be optimized in after deserialization.

**Pull request merged**